### PR TITLE
fix amqp for jessie

### PR DESCRIPTION
**Pull request by itscaro**
@mlocati I'm doing some further tests locally

libssl1.0-dev is present only in stretch
https://packages.debian.org/stretch/libssl1.0-dev

librabbitmq4 is not present in jessie, but librabbitmq-dev is
https://packages.debian.org/stretch/librabbitmq4
https://packages.debian.org/jessie/librabbitmq-dev

Humh is jessie supported anymore? zip is not installable in jessie, I don't use jessie anymore.

I tested on `7.1-cli-jessie` with `igbinary redis amqp apcu bcmath gettext gmp intl opcache pdo_dblib pdo_sqlite pdo_pgsql pgsql memcached imagick rdkafka shmop sockets sysvmsg sysvsem sysvshm wddx xml xsl zip ldap`
```
FAILED TO LIST THE WHOLE PACKAGE LIST FOR
librabbitmq4 libsybdb5 libpq5 libpq5 libmemcachedutil2 libmagickwand-6.q16-? librdkafka++1 libxslt1.1 libzip4 libmbedtls1?

COMMAND OUTPUT:
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package librabbitmq4
E: Unable to locate package libzip4
E: Unable to locate package libmbedtls1?
E: Couldn't find any package by regex 'libmbedtls1?'

E: apt-get failed
FAILED TO LIST THE WHOLE PACKAGE LIST FOR
librabbitmq-dev libssh-dev libgmp-dev libicu-dev freetds-dev libpq-dev libpq-dev libmemcached-dev zlib1g-dev libmagickwand-dev librdkafka-dev libxml2-dev libxslt-dev cmake gnutls-dev libssl-dev libzip-dev libbz2-dev libmbedtls-dev zlib1g-dev libldap2-dev

COMMAND OUTPUT:
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libmbedtls-dev

E: apt-get failed
### INSTALLING REQUIRED PACKAGES ###
# Packages to be kept after installation:
# Packages to be used only for installation:
```

### CODE REVIEW

**Comment by mlocati** [comment]
I added [this empty commit](https://github.com/mlocati/docker-php-extension-installer/pull/67/commits/ff7fd287fdbb1b52082a5e8297a0e899dcdbfdfc) containing `est: mongo, mongodb, zip, amqp+mongo, amqp+mongodb, amqp+zip, -STOP-`, so that we check the script

**Comment by mlocati** [comment]
Please remark that at the beginning of the `buildRequiredPackageLists` function we haven't called `apt-get update` yet, so you may only find actually installed packages with `apt-cache search`.

Maybe we could use your [`getDistroVersion`](https://github.com/mlocati/docker-php-extension-installer/pull/63/commits/821cc51fc2c88387506eec6108b225d794698ab5#diff-cf224b833cafcc4b824a1d1fa53f14c2R48-R58) function instead.

**Comment by mlocati** [comment]
I've just merged #69: we now test [many more docker images](https://travis-ci.org/mlocati/docker-php-extension-installer/builds/627666784).

**Comment by itscaro** [comment]
@mlocati rebased :)

File: install-php-extensions [hunk 1]
```diff
@@ -531,7 +560,14 @@ buildRequiredPackageLists () {
 				buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile cmake gnutls-dev libzip-dev libressl-dev zlib-dev"
 				;;
 			zip@debian)
-				buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libzip4 libmbedtls1?"
+				case "$(getDistroVersion)" in
+					debian8)
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libzip2"
```
Comment (itscaro) [inline install-php-extensions]: What should be replacement for `libmbedtls1?` in jessie?

File: install-php-extensions [hunk 2]
```diff
@@ -348,7 +370,14 @@ buildRequiredPackageLists () {
 				buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile icu-dev cyrus-sasl-dev snappy-dev libressl-dev zlib-dev"
 				;;
 			mongodb@debian)
-				buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libsnappy1v5"
+				case "$(getDistroVersion)" in
+					debian8)
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libsnappy1"
```
Comment (itscaro) [inline install-php-extensions]:
```
### REMOVING UNNEEDED PACKAGES ###
Reading package lists...
Building dependency tree...
Reading state information...
The following packages will be REMOVED:
  icu-devtools* libicu-dev* libicu52* libsasl2-dev* libsnappy-dev* libsnappy1*
  libssl-dev* libssl-doc* zlib1g-dev*
```

**Comment by mlocati** [comment]
Please open a pull request for every extension you fix

**Comment by mlocati** [comment]
PS: I just merged #70, where I added the new `getDistroVersion` function (slightly different than the one you originally wrote)

**Comment by mlocati** [comment]
PPS: I've just launched a local script that tests all the supported extensions for all the supported distributions.
Once it ends (it'll take hours), I'll submit a pull request for every extensions that needs some fixes.
Of course, in the meantime you can submit pull requests for the extensions that you know don't work (but please, one pull request per extension).

**Comment by mlocati** [comment]
What about https://github.com/mlocati/docker-php-extension-installer/pull/73 ?

File: install-php-extensions [hunk 3]
```diff
@@ -195,12 +195,15 @@ buildRequiredPackageLists () {
 			buildRequiredPackageLists_volatile="$PHPIZE_DEPS"
 			;;
 		debian)
-			if test -n "$(apt-cache search libssl1.0 | grep -E ^libssl1\.0)"; then
```
Comment (mlocati) [inline install-php-extensions]: Why is this change related to amqp?

**Comment by itscaro** [comment]
Replaced by #73

File: install-php-extensions (hunk 3 above)
Comment (itscaro) [reply install-php-extensions]: error at my side.
